git_machete/git_operations.py

```diff
@@ -437,10 +437,15 @@ def get_committer_unix_timestamp_by_revision(self, revision: AnyBranchName) -> i
 
     def get_inferred_remote_for_fetching_of_branch(self, branch: LocalBranchShortName) -> Optional[str]:
         # Since many people don't use '--set-upstream' flag of 'push', we try to infer the remote instead.
-        for remote in self.get_remotes():
-            if f"{remote}/{branch}" in self.get_remote_branches():
-                return remote
-        return None
+        remotes = self.get_remotes()
+        for remote in remotes:
+            self.fetch_remote(remote)  # NOT SURE IF FETCHING REMOTES IS REALLY NECESSARY
```

Comment:
Nopes! I wouldn't do it here under the hood, this might lead to unexpected fetches where we don't want them